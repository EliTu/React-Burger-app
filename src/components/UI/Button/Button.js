import React from 'react';
import styles from './Button.module.css';
import PropTypes from 'prop-types';

const Button = ({ children, handleClick, type, toolBarButton }) => {
	// CSS Modules styles:
	const { Button } = styles;

	return (
		<button
			onClick={handleClick}
			className={[Button, styles[type], styles[toolBarButton]].join(' ')}
		>
			{children}
		</button>
	);
};

Button.propTypes = {
	children: PropTypes.oneOfType([
		PropTypes.arrayOf(PropTypes.node),
		PropTypes.node,
	]).isRequired,
	handleClick: PropTypes.func.isRequired,
	type: PropTypes.string,
	toolBarButton: PropTypes.string,
};

export default Button;
