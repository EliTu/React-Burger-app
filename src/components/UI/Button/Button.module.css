@import '../../../variables.css';

.Button {
    width: 95%;
    font-size: 1.15rem;
    align-self: baseline;
    margin: 0.5rem auto;
    padding: 0.5rem;
    border: 1px solid var(--coloring-opaque-dark);
    border-radius: 5px;
    background-color: initial;
    cursor: pointer;
    outline: none;
    transition: 0.1s all;
}

.Confirm {
    color: var(--glow-green-opaque-lighter);
    border: 2px solid var(--glow-green-opaque);
}

.Confirm:hover {
    color: var(--coloring-light);
    background-color: rgba(37, 168, 28, 0.781);
}

.Danger {
    color: var(--invalid-red-light);
    border: 2px solid var(--invalid-red-dark);
}

.Danger:hover {
    color: rgb(228, 25, 25);
    background-color: rgba(231, 159, 159, 0.74);
}

.Neutral {
    color: var(--coloring-opaque-dark);
    border: 2px solid var(--disabled-dark);
}

.Neutral:hover {
    color: rgb(228, 222, 222);
    background-color: rgba(175, 174, 174, 0.74)
}

/* Mobile - SideDrawer toggle */
@media (max-width: 500px) {
    .MenuToggle {
        color: var(--coloring-bright-light);
        font-weight: 800;
        font-size: 1.5rem;
        height: 100%;
        width: 20%;
        align-self: flex-end;
        margin: 0;
        padding: 0 0.9rem;
        border-bottom: none;
    }
}