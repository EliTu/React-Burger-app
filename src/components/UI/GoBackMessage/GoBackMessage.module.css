@import '../../../variables.css';

.GoBackMessage {
    color: var(--invalid-red-dark);
    font-size: 1.2rem;
    padding: 0.5rem;
    width: 65%;
    margin: 3rem auto;
    padding: 2rem 1rem;
    text-align: center;
    border: 2px solid var(--coloring-opaque-dark);
    border-radius: 7px;
}

.GoBackMessage button {
    margin: 0.7rem 0;
}

/* Small screen */
@media only screen and (max-width: 500px) {
    .GoBackMessage {
        width: 90%;
    }
}