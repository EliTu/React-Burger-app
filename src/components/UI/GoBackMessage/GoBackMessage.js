import React from 'react';
import Button from '../Button/Button';
import styles from './GoBackMessage.module.css';
import { withRouter } from 'react-router';

export const GoBackMessage = ({ content, history }) => {
	// CSS Modules styles:
	const { GoBackMessage } = styles;

	const handleBackClick = () => history.goBack();

	return (
		<div className={GoBackMessage}>
			{content}
			<Button handleClick={handleBackClick} type="Danger">
				Go back
			</Button>
		</div>
	);
};

export default withRouter(GoBackMessage);
