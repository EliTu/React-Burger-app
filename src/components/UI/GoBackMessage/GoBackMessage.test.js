import React from 'react';
import { GoBackMessage } from './GoBackMessage';
import Button from '../../UI/Button/Button';
import { shallow } from 'enzyme';

describe('<GoBackMessage>', () => {
	let component = shallow(<GoBackMessage content="abc" history />);
	let buttonComp = component.find(Button);

	it('should render the component without errors', () => {
		expect(component).toMatchSnapshot();
		expect(component.length).toBe(1);
		expect(component.length).not.toBe(2);
	});

	it('should render a Button component', () => {
		expect(component.find(Button).length).toBe(1);
		expect(component.find(Button).length).not.toBe(2);
		expect(buttonComp.dive().text()).toBe('Go back');
	});

	it('should render the content prop', () => {
		expect(
			component
				.children()
				.first()
				.text()
		).toBe('abc');
	});
});
