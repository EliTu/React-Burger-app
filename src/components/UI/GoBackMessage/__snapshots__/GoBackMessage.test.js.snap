// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoBackMessage> should render the component without errors 1`] = `
<div
  className="GoBackMessage"
>
  abc
  <Button
    handleClick={[Function]}
    type="Danger"
  >
    Go back
  </Button>
</div>
`;
